gax-java/gax-httpjson/src/main/java/com/google/api/gax/httpjson/ProtoMessageRequestFormatter.java

```diff
@@ -161,9 +166,10 @@ public Builder<RequestT> setPath(
       return this;
     }
 
-    @BetaApi
-    public Builder<RequestT> setAdditionalPaths(String... rawAdditionalPaths) {
-      this.rawAdditionalPaths = Arrays.asList(rawAdditionalPaths);
+    public Builder<RequestT> setAdditionalPathsExtractorMap(
+        Map<PathTemplate, FieldsExtractor<RequestT, Map<String, String>>>
+            additionalPathsExtractorMap) {
+      this.additionalPathsExtractorMap = additionalPathsExtractorMap;
```

Comment:
Added this part as clean up for this file. It's marked as `@BetaApi` which may give us some flexibility in renaming/ updating this function.